lib/board.rb

```diff
@@ -0,0 +1,90 @@
+class Board
+  attr_accessor :grid 
+
+  def initialize(symbols = Array.new(9))
+    @grid = symbols
+  end
+
+  def is_empty
+    !(grid.include?(:X) or grid.include?(:O))
+  end
+
+  def update(index, symbol)
+    grid_before_index = copy_portion_of_grid(index)
+    grid_after_index = copy_portion_of_grid(grid.size - (index + 1))  
+
+    set_grid Array.new(grid_before_index + [symbol] + grid_after_index)
+  end
+
+  def has_free_spaces 
+    grid.include?(nil)
+  end
+
+  def has_winning_combination
+    all_winning_combinations = all_winning_combinations_for_3x3_grid
+
+    indices_of_every_x = grid.each_index.select { |v| grid[v] == :X} 
+    indices_of_every_o = grid.each_index.select { |v| grid[v] == :O} 
+
+    has_win_for_x = check_for_winning_combination_in(indices_of_every_x, all_winning_combinations)
+    has_win_for_o = check_for_winning_combination_in(indices_of_every_o, all_winning_combinations)
+
+    return (has_win_for_x or has_win_for_o)
+    #   return false
```

Comment:
You can see in the last commit (https://github.com/gemcfadyen/Apprenticeship-RubyTicTacToe/commit/cc23e217e5fadb47120986056f183728c043fa22) I removed this comment. However it is still showing here, even though locally I have no changes staged for commit.